exhale/deploy.py

```diff
@@ -109,8 +109,8 @@ def _generate_doxygen(doxygen_input):
             #
             # See excellent synopsis:
             # https://thraxil.org/users/anders/posts/2008/03/13/Subprocess-Hanging-PIPE-is-your-enemy/
-            _, tmp_out_path  = tempfile.mkstemp(prefix="exhale_launched_doxygen_buff")
-            _, tmp_err_path  = tempfile.mkstemp(prefix="exhale_launched_doxygen_buff")
+            tmp_out_path_fd, tmp_out_path  = tempfile.mkstemp(prefix="exhale_launched_doxygen_buff")
+            tmp_err_path_fd, tmp_err_path  = tempfile.mkstemp(prefix="exhale_launched_doxygen_buff")
             tmp_out_file     = codecs.open(tmp_out_path, "r+", "utf-8")  # read/write (read after communicate)
```

Comment:
I see, so the first thing returned is an already open file.  I think this means we can get rid of the `codecs.open` and just use `tmp_out_path_fd` and `tmp_err_path_fd` the whole time and get rid of `tmp_out_file` and `tmp_err_file`?

Reply:
Probably, I'm not expert at playing with file descriptors though, I'll try and have a look!

Reply:
OK, so we have 2 ways of doing it (example her for stdout):
```
import io
...
fd_out, tmp_out_path = tempfile.mkstemp(prefix="exhale_launched_doxygen_buff")
tmp_out_file = io.open(fd_out, 'r+', encoding='utf-8')
....
tmp_out_file.close()
os.remove(tmp_out_path)
```

or, much more concise:

```
f_out = tempfile.TemporaryFile('r+', encoding='utf-8', prefix='exhale_launched_doxygen_buff')
...
f_out.close()
```
(the file is automatically destroyed when explicitly or implicitly closed)

needless to say I prefer the 2nd solution (yes it works with all current versions of Python)

Reply:
Hmmm.  Ok I'll take a closer look, I recall seeing `TemporaryFile` but deciding against it for some reason.  I think it was permissions, but don't remember.

Reply:
Maybe it was because ``tempfile.TemporaryFile`` did not accept an ``encoding`` kwarg on python 2.7? (just found that out!)

Reply:
That definitely sounds familiar.